parameter/ElementHandle.cpp

```diff
@@ -41,6 +41,36 @@ using std::string;
 using std::mutex;
 using std::lock_guard;
 
+template <class T>
+struct isVector : std::false_type
+{
+};
+
+template <class T>
+struct isVector<std::vector<T>> : std::true_type
+{
+};
+
+template <class T>
+struct remove_ref_cv
+{
+    using type = typename std::remove_cv<typename std::remove_reference<T>::type>::type;
+};
+
+template <class T>
+static typename std::enable_if<not isVector<typename remove_ref_cv<T>::type>::value,
+                               size_t>::type getUserInputSize(T /*values*/)
+{
+    return 0;
+}
+
+template <class T>
+static typename std::enable_if<isVector<typename remove_ref_cv<T>::type>::value, size_t>::type
+getUserInputSize(T values)
+{
+    return values.size();
+}
+
```

Comment:
I'm not sure it is worth it: It works but only if I add an overload for std::string (it has a size method but we don't want to consider it as a collection). I can't guarantee, ahead of time, that no future "scalar" type will have a size() method. We would rely too much on future tests begin correctly written.